sympy/matrices/common.py

```diff
@@ -2082,20 +2083,26 @@ def _eval_add(self, other):
         return self._new(self.rows, self.cols,
                          lambda i, j: self[i, j] + other[i, j])
 
-    def _eval_matrix_mul(self, other, **kwargs):
+    def _eval_matrix_mul(self, other, mulsimp=None):
```

Comment:
I don't think we need to add this to MatrixArithmetic. In fact why is there another implementation of `_eval_matrix_mul` here anyway?

Reply:
From what I saw to support other matrix abstract matrix types which do not derive from Dense or Sparse. There is a MutablePolyDenseMatrix but that comes from Dense. There are tests for abstract matrices also I think the quantum stuff has some custom matrices. In modified it just in case some abstract matrix were to be multiplied with a Dense or Sparse.

Reply:
All these classes just seems like a mess to me. AFAICT the implementation here was deliberately minimal. I don't think we should change it, especially if it means duplicating the code. What is an example of how this code would be used?

Reply:
`_matrixify` which is used a bit uses `_MatrixWrapper` which is a `is_MatrixLike = True` which seems to be treated as abstract but I can not give you a concrete case as it does seem to be a mess.